### PR TITLE
Atmel SAM: Introduce SMC driver

### PR DESCRIPTION
**Pull request by pdgendt**
Add support for the ATSAM4SA16C variant and add a driver for the Atmel SMC/EBI module.

Tested on a custom board

### CODE REVIEW

**Comment by nandojve** [comment]
Hi @pdgendt ,

Thank you for this contribution. It seems a very clean implementation. I hope can review soon when merge window is open again.
I'm missing few things:

 * It is required a board in tree be configured with this driver to allow CI tests. Is it possible enable IS66WV51216DALL support at SAM4S, or even SAM4E? Please, let us know at least what is your use case.
 * It is necessary create a test. It doesn't need to be so elaborated but without it won't be possible build the driver at CI with a valid config. My suggestion is add as a sample at `samples/drivers/smc_<use case>` where _use case_ can be _sram_, _sdram_ etc.
 * It will be necessary add an entry as **smc** at board yaml file to enable twister tests. Be aware that **smc** will be the filter entry in the test case. https://github.com/zephyrproject-rtos/zephyr/blob/655b0d21274a5277d00091cf59c0d591ec0df5bb/boards/arm/sam4s_xplained/sam4s_xplained.yaml#L9-L16
 * I think SAM4S and SAM4E have same SMC IP, could you confirm? If yes, could you add SMC node for SAM4E too.
 * How users will mapped memory?

**Comment by pdgendt** [comment]
Hi @nandojve, thank you for the response. I've update the PR, please check my responses below.

> It is required a board in tree be configured with this driver to allow CI tests. Is it possible enable IS66WV51216DALL support at SAM4S, or even SAM4E? Please, let us know at least what is your use case.

I've added the `IS66WV51216DALL` to the sam4s_xplained board, however I don't have access to this HW, can you/someone test the changes made? I've taken IDs/addresses/timings from datasheets and example projects.

> It is necessary create a test. It doesn't need to be so elaborated but without it won't be possible build the driver at CI with a valid config. My suggestion is add as a sample at `samples/drivers/smc_<use case>` where _use case_ can be _sram_, _sdram_ etc.

Is this still required if it is added to an existing board where other tests will probably already cover this?

> It will be necessary add an entry as **smc** at board yaml file to enable twister tests. Be aware that **smc** will be the filter entry in the test case. https://github.com/zephyrproject-rtos/zephyr/blob/655b0d21274a5277d00091cf59c0d591ec0df5bb/boards/arm/sam4s_xplained/sam4s_xplained.yaml#L9-L16

Added.

> I think SAM4S and SAM4E have same SMC IP, could you confirm? If yes, could you add SMC node for SAM4E too.

Added based on the SAM4E datasheet, unable to test due to lack of HW.

> How users will mapped memory?

I've updated the `yaml` file on how to add the memory to the dts.

File: drivers/memc/memc_sam_smc.c
Comment (nandojve) [inline drivers/memc/memc_sam_smc.c:39]: I think all comments in this file could be dropped.

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @pdgendt ,

It will be necessary add a test at `tests/drivers/memc`. My suggestion is do something simple like write a pattern, read back and compare. This could be a generic test that may be used for any future `memc_sram` driver inclusive.

**Comment by pdgendt** [comment]
> It will be necessary add a test at `tests/drivers/memc`. My suggestion is do something simple like write a pattern, read back and compare. This could be a generic test that may be used for any future `memc_sram` driver inclusive.

Hey @nandojve,

I was looking into this, and there's a test for `stm32_sdram` which could be reused. However this requires some more effort for the linker sections to be more generic.

Looking at #42008 and #40422 we could rework that existing test once those PRs are merged.

File: boards/arm/sam4s_xplained/sam4s_xplained_defconfig
Comment (nandojve) [reply boards/arm/sam4s_xplained/sam4s_xplained_defconfig]: Hi @pdgendt ,

I was looking closely about this and I agreed with you. Let's keep it!

File: drivers/memc/CMakeLists.txt
Comment (nandojve) [reply drivers/memc/CMakeLists.txt:12]: Yes, correct! Let's keep as is. My initial thought is that SMC would be same controller that handles SDRAM, which is not the case.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (nandojve) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:8]: Use the full list: SRAM, PSRAM, PROM, EPROM, EEPROM, LCD Module, NOR Flash and NAND Flash.

Drop SDRAM because there is a dedicated controller to handle that.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (nandojve) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:65]: The EBI allows you to have SDRAM but uses a dedicated controller which is not SMC. Please, fix the example.

**Review by nandojve — CHANGES_REQUESTED** [review]
> I was looking into this, and there's a test for stm32_sdram which could be reused. However this requires some more effort for the linker sections to be more generic.

If it is possible, better! Project prefers reuse instead add new ones.

> Looking at https://github.com/zephyrproject-rtos/zephyr/pull/42008 and https://github.com/zephyrproject-rtos/zephyr/pull/40422 we could rework that existing test once those PRs are merged.

Your PR will be conditioned to be merged after those.

I was looking this controller and my initial thought was that SMC should handle SDRAM too. After looking datasheets I noted that SAM uses a dedicated controller for SDRAM. So, in this case, we have two distinct drivers and those are independently.

On that research I noted that SAMS7/E7/V7 uses same SMC ip. It is up to you add that nodes like SAM4E. There is a minimal change in the driver that could be added which will allow SAMS7/E7/V7 have 16 bit access instead only 8. If you don't mind add this, you can upstream the whole feature set. This is a plus not a block from my side.

**Comment by pdgendt** [comment]
> > Looking at #42008 and #40422 we could rework that existing test once those PRs are merged.
> 
> Your PR will be conditioned to be merged after those.
> 

So we wait a bit, or?

> I was looking this controller and my initial thought was that SMC should handle SDRAM too. After looking datasheets I noted that SAM uses a dedicated controller for SDRAM. So, in this case, we have two distinct drivers and those are independently.

Do you mean on the SMC module as well?

> On that research I noted that SAMS7/E7/V7 uses same SMC ip. It is up to you add that nodes like SAM4E. There is a minimal change in the driver that could be added which will allow SAMS7/E7/V7 have 16 bit access instead only 8. If you don't mind add this, you can upstream the whole feature set. This is a plus not a block from my side.

I currently have limited time to add these additional SoCs, and no hardware to test. The SAM4E looked easy, but I also have no way to test this. I guess if someone needs them, is should be easy to test and upstream.

**Comment by nandojve** [comment]
> > Your PR will be conditioned to be merged after those.
> 
> So we wait a bit, or?

Yes, to reuse.

> Do you mean on the SMC module as well?

As far I understood SMC not support SDRAM. SDRAM is handled by a dedicated controller. At end, EBI is a set of one or two controllers (SMC + SDRAM).

**Comment by nandojve** [comment]
Hi @pdgendt ,

I would like to add this after #43731.

**Comment by mbolivar-nordic** [comment]
Not for v3.1

**Comment by nandojve** [comment]
Hi @pdgendt ,

The #43731 is already in and you can update PR with the new pinctrl API support.

**Comment by pdgendt** [comment]
> The #43731 is already in and you can update PR with the new pinctrl API support.

Hi @nandojve,

I've updated the PR to use the new pinctrl API, however I don't have the board to verify, is this something you could do?

Thanks

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @pdgendt ,

Yes, I can test in the boards. Please, make sure that test/drivers/memc will trigger. Currently it is not.

**Comment by nandojve** [comment]
Hi @pdgendt ,

Implementation is OK, tested on sam4s_xplained

```console
*** Booting Zephyr OS build zephyr-v3.1.0-541-ga7b1463ba7aa  ***
Running TESTSUITE smc_sram
===================================================================
START - test_sram1_rw
 PASS - test_sram1_rw in 0.1 seconds
===================================================================
START - test_sram2_rw
 PASS - test_sram2_rw in 0.1 seconds
===================================================================
TESTSUITE smc_sram succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

I was wondering if it is a good idea unify those tests.

**Comment by pdgendt** [comment]
> Implementation is OK, tested on sam4s_xplained

@nandojve Nice, thanks for testing!

> I was wondering if it is a good idea unify those tests.

I think that would be best, indeed. But maybe this could be done in a separate PR?

**Review by nandojve — DISMISSED** [review]
Hi @pdgendt,

> I think that would be best, indeed. But maybe this could be done in a separate PR?

It is OK from my side.

Thank you!

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (henrikbrixandersen) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:48]: Is the label really required?

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (henrikbrixandersen) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:65]:
```suggestion
    Child device nodes are representing devices connected to the EBI/SMC bus.
```

File: tests/drivers/memc/smc_sram/testcase.yaml
Comment (henrikbrixandersen) [inline tests/drivers/memc/smc_sram/testcase.yaml:5]: Another option would be to put this under `tests/boards/atmel_sam` or similar - of make it generic for all memc drivers.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: tests/drivers/memc/smc_sram/testcase.yaml
Comment (pdgendt) [reply tests/drivers/memc/smc_sram/testcase.yaml:5]: This suggestion was already done by @nandojve , but I wanted to this in a separate PR.

Comment (henrikbrixandersen) [reply tests/drivers/memc/smc_sram/testcase.yaml:5]: Why not fix it before introducing it?

Comment (gmarull) [reply tests/drivers/memc/smc_sram/testcase.yaml:5]: note: `memc` are a set of drivers without an API, so a generic test isn't possible afaik.

Comment (henrikbrixandersen) [reply tests/drivers/memc/smc_sram/testcase.yaml:5]: The current test simply tests RW access to a memory region from a device on the memory controller - that sounds like something that could work for at least more than one memc device type.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (pdgendt) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml]: Not for me, I compared with other similar drivers, and it appears this is added for most.
Will remove.

File: tests/drivers/memc/smc_sram/testcase.yaml
Comment (pdgendt) [reply tests/drivers/memc/smc_sram/testcase.yaml:5]: @henrikbrixandersen ok, to update the test in a follow-up PR?

**Comment by pdgendt** [comment]
Rebased and added multi-instance support.

File: tests/drivers/memc/smc_sram/testcase.yaml
Comment (henrikbrixandersen) [reply tests/drivers/memc/smc_sram/testcase.yaml:5]: I don't see why we should introduce the test with identified issues only to do a follow-up PR. Please address it in this PR.

Comment (pdgendt) [reply tests/drivers/memc/smc_sram/testcase.yaml]: Done

**Comment by pdgendt** [comment]
@nandojve could you re-test the new implementation with the `xplained` board?
Thanks!

**Review by henrikbrixandersen — DISMISSED** [review]
Thanks :)

**Comment by nandojve** [comment]
> @nandojve could you re-test the new implementation with the `xplained` board? Thanks!

Hi @pdgendt ,
I`m without HW this week. I can run tests after July 20th when I return from a trip.
Overall LGTM!

**Comment by carlescufi** [comment]
@nandojve can you take a look?

**Comment by pdgendt** [comment]
> Hi @pdgendt , I`m without HW this week. I can run tests after July 20th when I return from a trip. Overall LGTM!

Hey @nandojve , friendly reminder to revisit this :)

File: dts/arm/atmel/sam4e.dtsi
Comment (galak) [inline dts/arm/atmel/sam4e.dtsi:264]: remove label prop

File: dts/arm/atmel/sam4s.dtsi
Comment (galak) [inline dts/arm/atmel/sam4s.dtsi:222]: remove label prop

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (galak) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:66]: Does `reg` here represent a chip select or something?

File: drivers/memc/Kconfig.sam
Comment (galak) [inline drivers/memc/Kconfig.sam:2]: Update Kconfig to new pattern based on autogenerated <BUILD_DIR>/Kconfig/Kconfig.dts.  See https://github.com/zephyrproject-rtos/zephyr/pull/48169 for example of the change.

**Review by galak — DISMISSED** [review]
Some minor Kconfig & dts changes.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (pdgendt) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:66]: Yes, it's a chip select.

Comment (galak) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:66]: maybe add a `description` field to `reg` and mention that there.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (galak) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:126]: remove white space.  Also, not sure how `fields` NWE/NRD and an array quite make sense together.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (galak) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:75]: Do we need `default:` for all these properties?  What are these default values based on?

**Review by galak — COMMENTED** [review]
some comments on the binding.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (pdgendt) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:75]: The default values are based on the datasheet, the register data after reset.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (pdgendt) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:126]: I used the array type to create a compact view instead of an entry for each item, and the values are set into the same HW register.

Comment (galak) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:126]: The array is fine, I just find the wording confusing.  not sure if `[NWE NRD]` in the description would make it a little clearer.

File: tests/drivers/memc/ram/testcase.yaml
Comment (nandojve) [inline tests/drivers/memc/ram/testcase.yaml:9]: twister is skipping `sam4s_xplained` board. It should be possible build locally with, I think:
`./scripts/twister -p sam4s_xplained -t tests/drivers/memc/ram`

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @pdgendt ,

I run on board and all tests passed, see below:
```
*** Booting Zephyr OS build zephyr-v3.1.0-2551-g2faab4602eff  ***
Running TESTSUITE test_ram
===================================================================
START - test_sdram1
 SKIP - test_sdram1 in 0.1 seconds
===================================================================
START - test_sdram2
 SKIP - test_sdram2 in 0.1 seconds
===================================================================
START - test_sram1
 PASS - test_sram1 in 0.1 seconds
===================================================================
START - test_sram2
 PASS - test_sram2 in 0.1 seconds
===================================================================
TESTSUITE test_ram succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

Please, check why twister is skipping the test.

File: tests/drivers/memc/ram/testcase.yaml
Comment (pdgendt) [reply tests/drivers/memc/ram/testcase.yaml:9]: Shouldn't this work? I'm not sure how to fix/improve this.
I thought about dropping the `dt_compat_enabled` filter entirely and check the nodelabels instead:

```yaml
tests:
  drivers.memc.ram:
    tags: drivers memc
    depends_on: memc
    filter: dt_nodelabel_enabled("sdram1") or dt_nodelabel_enabled("sdram2") or
            dt_nodelabel_enabled("sram1") or dt_nodelabel_enabled("sram2")

```

Any suggestions?

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (mbolivar-nordic) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:116]: All of these defaults must be justified in the descriptions.  https://docs.zephyrproject.org/latest/build/dts/bindings.html#rules-for-default-values

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]

File: tests/drivers/memc/ram/testcase.yaml
Comment (nandojve) [reply tests/drivers/memc/ram/testcase.yaml:9]: No clue about it. Please, take a look at Discord -> Infrastructure channel.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (pdgendt) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml]: @mbolivar-nordic I've updated the descriptions with information from the datasheet

File: tests/drivers/memc/ram/testcase.yaml
Comment (pdgendt) [reply tests/drivers/memc/ram/testcase.yaml:9]: @nandojve I've added `sam4s_xplained` as an integration platform, CI should build it now.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (mbolivar-nordic) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:55]: Is this actually going into a register somewhere or are you just using it to work around hal limitations? If the former, it's OK, but if the latter, then @galak has a longstanding prohibition against such properties.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (mbolivar-nordic) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:101]: This is kind of confusing still. It says 1 is the default but the default is actually an array. I guess the description is meant to say the first element is NWE setup timing, the second is NCS_WR, etc.? I personally find that a bit confusing if so.

My bigger question here though is why are you assuming devices that perform actions on rising edges? Would a different configuration only appear under very unusual conditions? If it's common for devices not to perform actions on rising edges, then these should be made required instead.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Still keeping a -1 on for now because it's not clear to me if this is a correct use of default.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (pdgendt) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:55]: The `peripheral-id`? This is how all the `sam` SoCs identify a HW block, and isn't something new. I don't want to turn this PR into something bigger.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (pdgendt) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:101]: I was trying to translate the datasheet a bit into the description and I thought it would be safer to have a minimal setup timing as the default for all read/write modes.
image

Maybe our atmel/michrochip maintainer @nandojve can suggest how the (timing) properties should be set?

Sorry this has taken already so many iterations, but I agree it has to be clear for everyone.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (nandojve) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:55]: Peripheral-is represents a bit value used to enable clock on the peripheral. For each SoC model this value change the current implementation it is the easy solution to allow share same driver between SoCs and allow DT_INST_FOREACH_STATUS_OKAY for instantiate. I think this property could be removed in the future after implement a clock driver and clock tree.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (nandojve) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:101]: Hi @pdgendt ,

The devicetree properties must be well documented. A maintainer from devicetree should read the documentation and have a very clear picture about it without inspect code or even datasheet. Otherwise no one can provide fixes in some situations and it is even more important when the property have a **default** value. I think both @mbolivar-nordic and @galak are arguing in that direction to make sure that properties are well defined and well documented.

In regards of timing parameters, I'm OK to have it as a array. I think what @mbolivar-nordic is asking to you go in the direction of have 100% sure that a  `default` is fine for this driver since the documented is not 100% clear. The default should be used only when you have a generic case that can full fill majority of scenarios otherwise `required` field should be added and set as `true`. This imposes that each board must explicit define the values since no `default` property will be available. The other think that I noted is that for someone that is not aware about SoC it is difficult to understand the default values.

If you look at my current PR #48377 you will noted that I'm in the same situation. The amount of information that I added at first push wasn't enough to make it clear. A second version was send with substantial more documentation and examples. This kind of situation happen because, in your case, the driver is one of the first and maintainers will look in deep to make sure it is clear how properties were define. Remember that people inspect code to pick ideas for drivers and this maybe be a reference for others.

Questions that may help to evaluate the properties:

* What happen if user selects another SDRAM manufacturer? Are the parameters same?
* What happen if user selects a slower speed memory? The current default configuration is set up for the slowest case?
* The current default values may work for all memories or at least for the majority of that?
* If I set up SAM4E-XPRO board will be possible to use same SDRAM parameters from SAM4S-Xplained (since memories are different)?

Sugestions:

 * Increase the amount of documentation
 * Add the sam4s_xplained configuration as an example in the devicetree documentation
 * Explain how to configure each parameter and why you select that value since it have a relationship with memory used and the SoC controller.

Comment (pdgendt) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml]: Thanks @nandojve for the elaborate response.

Based on the feedback, it is clear that default values do not make sense, due to the amount of different external controllers possible.
All properties are now required without defaults.

Is the `sam4s_xplained` example configuration in the devicetree documentation sufficient?

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @pdgendt ,

I think it is better but still missing important information. I tried highlight few points to you.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (nandojve) [inline dts/bindings/memory-controllers/atmel,sam-smc.yaml:127]: This value is used to effectivelly read/write at memory region (pulse phase).
The pulse value is an array of the signals NWE, NCS_WR, NRD and NCS_RD were
each value is configured in terms of MCK cycles and value of 0 is forbiden.
Each value is encoded in 7 bits where the highest bit adds an offset of 256
cycles. The effective value for each element is: 256 x setup[6] + setup[5:0]

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @pdgendt ,

It still not clear. Please, see my suggestions and adapt as you wish.

**Comment by pdgendt** [comment]
Thanks @nandojve, for all your efforts to help out!

I updated the suggestions you made, but adjusted the example timings to the datasheet specifications.
A minimum setup time of 5ns and puls time of 45ns where swapped in your description?

**Comment by nandojve** [comment]
Hi @pdgendt ,

I`m OK with changes. Let`s ask to @mbolivar-nordic to check if fits.

BTW, we need fix flash stuff

**Comment by nandojve** [comment]
> BTW, we need fix flash stuff

Fixed by #48762

**Comment by nandojve** [comment]
Hi @mbolivar-nordic ,

No more CI erros and @pdgendt made few changes. I was wondering if you can revisit this PR.

File: dts/bindings/memory-controllers/atmel,sam-smc.yaml
Comment (mbolivar-nordic) [reply dts/bindings/memory-controllers/atmel,sam-smc.yaml:78]: > I think this property could be removed in the future after implement a clock driver and clock tree.

@nandojve based on our offline discussion, it sounds like you're agreeing to take this on so that these properties can be removed treewide for SAM SoCs. Given maintainer commitment to improve this, I am OK with this for now.

**Review by mbolivar-nordic — APPROVED** [review]
@pdgendt thanks for the big improvements. I'm glad to see the properties go to `required`. The `default` feature has its uses, but it is easy to abuse and I'm glad for your and @nandojve figuring out a better way forward.

**Review by nandojve — APPROVED** [review]

**Comment by talih0** [comment]
@pdgendt, @nandojve, I'm working on a similar driver for Infineon XMC4xxx.
A problem I've come across is that it's not possible to use zephyr_code_relocate() to move data, bss to external RAM with these type of drivers.

The driver set up happens with POST_KERNEL priority, but 
bss_zeroing_relocation(); and  data_copy_xip_relocation() are called very early at boot up. This driver should have the same problem. 

Did you come across something like this?

**Comment by pdgendt** [comment]
> @pdgendt, @nandojve, I'm working on a similar driver for Infineon XMC4xxx. A problem I've come across is that it's not possible to use zephyr_code_relocate() to move data, bss to external RAM with these type of drivers.
> 
> The driver set up happens with POST_KERNEL priority, but bss_zeroing_relocation(); and data_copy_xip_relocation() are called very early at boot up. This driver should have the same problem.
> 
> Did you come across something like this?

Hey @talih0, I did not need relocating to external RAM, so I can't really share my experience there. If this is something you want to be looked at, I would suggest creating a new issue. Comments on merged PRs are easily overlooked :)

**Comment by talih0** [comment]
thanks @pdgendt for following up, I'll create an issue.

**Pull request merged**
